cg/apps/crunchy.py

```diff
@@ -0,0 +1,320 @@
+        )
+
+        sbatch_content = sbatch_header + "\n" + sbatch_body
+        self._submit_sbatch(sbatch_content=sbatch_content, dry_run=dry_run)
+
+    # def spring(
+    #     self, fastq_first_path: Path, fastq_second_path: Path, dry_run: bool = False
+    # ):
+    #     """
+    #         Use spring to compress fastq_files
+    #     """
+    #
+    #     spring_path = self.change_suffix_spring(
+    #         fastq_first_path=fastq_first_path, fastq_second_path=fastq_second_path
+    #     )
+    #     job_name = spring_path.name + "_spring"
+    #     flag_path = self.get_flag_path(file_path=spring_path)
+    #
+    #     sbatch_header = self.get_slurm_header(
+    #         job_name=job_name,
+    #         account=self.slurm_account,
+    #         log_dir=self.slurm_log_dir,
+    #         crunchy_env=self.crunchy_env,
+    #     )
+    #
+    #     sbatch_body = self.get_slurm_spring(
+    #         fastq_first_path=fastq_first_path,
+    #         fastq_second_path=fastq_second_path,
+    #         spring_path=spring_path,
+    #         flag_path=flag_path,
+    #     )
+    #
+    #     sbatch_content = sbatch_header + "\n" + sbatch_body
+    #     self._submit_sbatch(sbatch_content=sbatch_content, dry_run=dry_run)
+
+    def _submit_sbatch(self, sbatch_content: str, dry_run: bool = False):
+        """Submit slurm job"""
+        if not dry_run:
+            with tempfile.NamedTemporaryFile(mode="w+t") as sbatch_file:
+
+                sbatch_file.write(sbatch_content)
+                sbatch_file.flush()
+                sbatch_parameters = [sbatch_file.name]
+                self.process.run_command(sbatch_parameters)
+                LOG.info(self.process.stderr)
+                LOG.info(self.process.stdout)
+        else:
+            LOG.info("Would submit following to slurm:\n\n%s", sbatch_content)
+
+    def cram_compression_done(self, bam_path: Path) -> bool:
+        """Check if CRAM compression already done for BAM file"""
+        cram_path = self.get_cram_path_from_bam(bam_path)
+        flag_path = self.get_flag_path(file_path=cram_path)
+
+        if not cram_path.exists():
+            LOG.info("No cram-file for %s", bam_path)
+            return False
+        if not self.get_index_path(cram_path).exists():
+            LOG.info("No index-file for %s", cram_path)
+            return False
+        if not flag_path.exists():
+            LOG.info("No %s file for %s", FLAG_PATH_SUFFIX, cram_path)
+            return False
+        return True
+
+    def bam_compression_possible(self, bam_path: Path) -> bool:
+        """Check if it CRAM compression for BAM file is possible"""
+        if bam_path is None or not bam_path.exists():
+            LOG.warning("Could not find bam %s", bam_path)
+            return False
+        if self.cram_compression_done(bam_path=bam_path):
+            LOG.info("cram compression already exists for %s", bam_path)
+            return False
+        return True
+
+    #
```

Comment:
another branch maybe?

Reply:
I think not done things should not go into production, I still thinks we should remove this dangling outcommented code, what do you say?